### PR TITLE
Use tcb::span in local_to_global and global_to_local

### PR DESCRIPTION
**Pull request by IgorBaratta**

### CODE REVIEW

File: cpp/dolfinx/common/IndexMap.cpp [hunk 1]
```diff
@@ -446,11 +446,12 @@ const std::vector<std::int64_t>& IndexMap::ghosts() const noexcept
   return _ghosts;
 }
 //-----------------------------------------------------------------------------
-void IndexMap::local_to_global(const std::int32_t* local, int n,
-                               std::int64_t* global) const
+void IndexMap::local_to_global(tcb::span<const std::int32_t> local,
```
Comment (garth-wells) [inline cpp/dolfinx/common/IndexMap.cpp:449]: Use `const` references here - save creating a new span if it's not needed.

Comment (IgorBaratta) [reply cpp/dolfinx/common/IndexMap.cpp]: done!

Comment (garth-wells) [reply cpp/dolfinx/common/IndexMap.cpp]: I think you can make both arguments const references.

Comment (chrisrichardson) [reply cpp/dolfinx/common/IndexMap.cpp]: span is very light, just 2 x int64 - is it worth taking a const ref?

**Review by garth-wells — APPROVED** [review]

**Review by chrisrichardson — APPROVED** [review]

**Pull request merged**